avocado/core/loader.py

```diff
@@ -684,6 +690,77 @@ def _find_avocado_tests(self, path, class_name=None):
                                                                       cl_tags),
                                     'path': path}
                             result[statement.name] = info
+                            continue
+
+                # Looking for the 'recursive' docstring or a class_name (what
+                # means we are under recursion)
+                if (safeloader.check_docstring_directive(docstring, 'recursive')
+                   or class_name is not None):
+                    info = {'methods': self._get_methods_info(statement.body,
+                                                              cl_tags),
+                            'path': path}
+                    result[statement.name] = info
+
+                    # Getting the list of parents of the current class
+                    parents = statement.bases
+
+                    # Searching the parents in the same module
+                    for parent in parents[:]:
+                        # Looking for a 'class FooTest(module.Parent)'
+                        if isinstance(parent, ast.Attribute):
+                            parent_class = parent.attr
+                        # Looking for a 'class FooTest(Parent)'
+                        else:
+                            parent_class = parent.id
+
+                        res = self._find_avocado_tests(path, parent_class)
+                        if res:
+                            parents.pop(0)
```

Comment:
You don't want to remove 0-st item, but the current item, therefor `parents.remove(parent)`.